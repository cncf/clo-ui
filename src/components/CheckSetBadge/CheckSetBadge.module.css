.badge {
  background-color: var(--bs-white);
  border-color: var(--clo-secondary);
  font-size: 0.7rem;
  padding: 0.2em 0.5em;
  height: 19px;
  line-height: 1.15;
}

.symbol {
  margin-top: -2px;
}
