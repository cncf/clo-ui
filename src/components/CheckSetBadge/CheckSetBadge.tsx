import React from 'react';

import styles from './CheckSetBadge.module.css';

export enum CheckSet {
  Code = 'code',
  CodeLite = 'code-lite',
  Community = 'community',
  Docs = 'docs',
}

export interface ICheckSetBadgeProps {
  checkSets: CheckSet[];
  className?: string;
}

export const CheckSetBadge = (props: ICheckSetBadgeProps) => {
  return (
    <div className={`d-inline-flex ${props.className}`}>
      <div
        data-testid="repo-kind-badge"
        className={`badge text-secondary border border-secondary rounded-0 position-relative text-uppercase ${styles.badge}`}
      >
        <div className="d-flex flex-row align-items-center">
          {props.checkSets.map((k: CheckSet, index: number) => {
            return (
              <React.Fragment key={`kind_${k}`}>
                {index !== 0 && <div className={`px-1 position-relative ${styles.symbol}`}>+</div>}
                {k}
              </React.Fragment>
            );
          })}
        </div>
      </div>
    </div>
  );
};
