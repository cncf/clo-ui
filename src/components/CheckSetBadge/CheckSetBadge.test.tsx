import { render, screen } from '@testing-library/react';

import { CheckSet, CheckSetBadge } from './CheckSetBadge';

const defaultProps = {
  checkSets: ['code', 'community'] as CheckSet[],
};

describe('CheckSetBadge', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<CheckSetBadge {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<CheckSetBadge {...defaultProps} />);

    const badge = screen.getByTestId('checkset-badge');
    expect(badge).toHaveTextContent('code+community');
  });
});
