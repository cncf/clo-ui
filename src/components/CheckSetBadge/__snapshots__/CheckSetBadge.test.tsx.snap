// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckSetBadge creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-inline-flex undefined"
  >
    <div
      class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase badge"
      data-testid="checkset-badge"
    >
      <div
        class="d-flex flex-row align-items-center"
      >
        code
        <div
          class="px-1 position-relative symbol"
        >
          +
        </div>
        community
      </div>
    </div>
  </div>
</DocumentFragment>
`;
